nservicebus/azure-storage-queues/configuration.md

```diff
@@ -17,7 +19,11 @@ The Azure Storage Queues Transport can be configured using the following paramet
  * `MaximumWaitTimeWhenIdle`: In order to save money on the transaction operations, the transport optimizes wait times according to the expected load. The transport will back off when no messages can be found on the queue. The wait time will be increased linearly, but it will never exceed the value specified here, in milliseconds. The default value is 1000 (i.e. 1 second).
  * `PurgeOnStartup`: Instructs the transport to remove any existing messages from the input queue on startup. The default value is `false`, i.e. messages are not removed when endpoint starts.
  * `MessageInvisibleTime`: The [visibilitytimeout mechanism](https://msdn.microsoft.com/en-us/library/azure/dd179474.aspx), supported by Azure Storage Queues, causes the message to become *invisible* after read for a specified period of time. If the processing unit fails to delete the message in the specified time, the message will reappear on the queue. Then another process can retry the message. The default value is 30000, in milliseconds (i.e. 30 seconds).
- * `BatchSize`: The number of messages that the transport tries to pull at once from the storage queue. The default value is 10. Depending on the expected load, the value should vary between 1 and 1000 (the maximum).
+ * `BatchSize`: The number of messages that the transport tries to pull at once from the storage queue. The default value is 10 in Version 6 and below and 32 in Version 7. Depending on the expected load, the value should vary between 1 and 32 (the maximum).
+ * `DegreeOfReceiveParallelism`: The number of parallel receive operations that the transport is issuing against the storage queue to pull messages out of it. By default this value is dynamically set by the endpoints concurrency limit. The algorithm is the following:
+ `Degree of parallelism = square root of MaxConcurrency (rounded)`. Here are a few examples: `(concurrency 1 = 1, concurrency 10 = 3, concurrency 20 = 4, concurrency 50 = 7, concurrency 100 [default] = 10, concurrency 200 = 14, concurrency 1000 = 32 [maximum])`. This means ten message processing loops will receive up to the configured `BatchSize` number of messages in parallel. For example with the default `BatchSize` of 32 and the default degree of parallelism of 10 the transport will be able to receive 320 messages from the storage queue at the same time.
+
+ WARNING: You need to carefully select the values of `BatchSize` , `DegreeOfParallelism`, `Concurrency`, [ServicePointManager Settings](/nservicebus/azure-storage-persistence/performance-tuning.md) and the other parameters like `MaximumWaitTimeWhenIdle` in order to get the desired speed out of the transport while not exceeding [the boundaries](https://azure.microsoft.com/en-us/documentation/articles/azure-subscription-service-limits/#storage-limits) of the allowed number of operations per second.
```

Comment:
That's what I did. even added it as related doc

On Wednesday, April 20, 2016, Szymon Kulec <notifications@github.com> wrote:

> In nservicebus/azure-storage-queues/configuration.md
> <https://github.com/Particular/docs.particular.net/pull/1432#discussion_r60393400>
> :
>
> > @@ -17,7 +19,11 @@ The Azure Storage Queues Transport can be configured using the following paramet
> >   * `MaximumWaitTimeWhenIdle`: In order to save money on the transaction operations, the transport optimizes wait times according to the expected load. The transport will back off when no messages can be found on the queue. The wait time will be increased linearly, but it will never exceed the value specified here, in milliseconds. The default value is 1000 (i.e. 1 second).
> >   * `PurgeOnStartup`: Instructs the transport to remove any existing messages from the input queue on startup. The default value is `false`, i.e. messages are not removed when endpoint starts.
> >   * `MessageInvisibleTime`: The [visibilitytimeout mechanism](https://msdn.microsoft.com/en-us/library/azure/dd179474.aspx), supported by Azure Storage Queues, causes the message to become *invisible* after read for a specified period of time. If the processing unit fails to delete the message in the specified time, the message will reappear on the queue. Then another process can retry the message. The default value is 30000, in milliseconds (i.e. 30 seconds).
> > - * `BatchSize`: The number of messages that the transport tries to pull at once from the storage queue. The default value is 10. Depending on the expected load, the value should vary between 1 and 1000 (the maximum).
> > + * `BatchSize`: The number of messages that the transport tries to pull at once from the storage queue. The default value is 10 in Version 6 and below and 32 in Version 7. Depending on the expected load, the value should vary between 1 and 32 (the maximum).
> > + * `DegreeOfReceiveParallelism`: The number of parallel receive operations that the transport is issuing against the storage queue to pull messages out of it. By default this value is dynamically set by the endpoints concurrency limit. The algorithm is the following:
> > + `Degree of parallelism = square root of MaxConcurrency (rounded)`. Here are a few examples: `(concurrency 1 = 1, concurrency 10 = 3, concurrency 20 = 4, concurrency 50 = 7, concurrency 100 [default] = 10, concurrency 200 = 14, concurrency 1000 = 32 [maximum])`. This means ten message processing loops will receive up to the configured `BatchSize` number of messages in parallel. For example with the default `BatchSize` of 32 and the default degree of parallelism of 10 the transport will be able to receive 320 messages from the storage queue at the same time.
> > +
> > + WARNING: You need to carefully select the values of `BatchSize` , `DegreeOfParallelism`, `Concurrency`, [ServicePointManager Settings](/nservicebus/azure-storage-persistence/performance-tuning.md) and the other parameters like `MaximumWaitTimeWhenIdle` in order to get the desired speed out of the transport while not exceeding [the boundaries](https://azure.microsoft.com/en-us/documentation/articles/azure-subscription-service-limits/#storage-limits) of the allowed number of operations per second.
>
> Shouldn't we just link to another doc describing tuning?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/1432/files/a0c5ff1f426e3d2a882d127ccb3f8eb72d55fa69#r60393400>
>